cpp/src/arrow/util/decimal_test.cc

```diff
@@ -1119,6 +1137,35 @@ class TestDecimalToReal : public ::testing::Test {
     // 2**64 + 2**41 (exactly representable in a float)
     CheckDecimalToReal<Decimal, Real>("18446746272732807168", 0, 1.8446746e+19f);
     CheckDecimalToReal<Decimal, Real>("-18446746272732807168", 0, -1.8446746e+19f);
+
+    // Integers are always exact
+    auto scale = Decimal::kMaxScale - 1;
+    std::string seven = "7.";
+    for (int32_t i = 0; i < scale; ++i) {
+      seven += "0";
+    }
```

Comment:
Can probably write this more tersely:
```suggestion
    seven.append(scale, '0');  // pad with trailing zeros
```